easybuild/easyblocks/o/openfoam.py

```diff
@@ -197,19 +213,28 @@ def configure_step(self):
         if LooseVersion(self.version) < LooseVersion("2") or openfoam_extend_v3:
             self.log.debug("List of deps: %s" % self.cfg.dependencies())
             for dep in self.cfg.dependencies():
-                self.cfg.update('prebuildopts', "%s_SYSTEM=1" % dep['name'].upper())
-                self.cfg.update('prebuildopts', "{0}_LIB_DIR=$EBROOT{0}/lib".format(dep['name'].upper()))
-                self.cfg.update('prebuildopts', "{0}_INCLUDE_DIR=$EBROOT{0}/include".format(dep['name'].upper()))
+                cur_dep = {
+                    'name': dep['name'].upper(),
+                    'root': get_software_root(dep['name'])
+                }
+                dep_vars = [
+                    ("%(name)s_SYSTEM", "1"),
+                    ("%(name)s_DIR", "%(root)s"),
+                    ("%(name)s_BIN_DIR", "%(root)s/bin"),
+                    ("%(name)s_LIB_DIR", "%(root)s/lib"),
+                    ("%(name)s_INCLUDE_DIR", "%(root)s/include"),
+                ]
+                for var, val in dep_vars:
+                    env.setvar(var % cur_dep, val % cur_dep)
```

Comment:
this is a bit weird, since you never need `%(name)s` in `val`, or `%(root)s` in `var`, so don't use `cur_dep`?